File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    queue<int> q;
    
    for(int i=0; i<progresses.size(); i++) {
        int day = 1;
        while(progresses[i] + day * speeds[i] < 100) {
            day++;
        }
        q.push(day);
    }
    
    while(!q.empty()) {
        int day = q.front();
        q.pop();
        int cnt = 1;
        
        while(!q.empty() && q.front() <= day) {
            q.pop();
            cnt++;
        }
        
        answer.push_back(cnt);
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        Queue<Integer> q = new LinkedList<>();
        
        for(int i=0; i<progresses.length; i++) {
            int progress = progresses[i];
            int speed = speeds[i];
            int j=0;
            while(progress < 100) {
                progress += speed;
                j++;
            }
            q.offer(j);
            
        }
        
        while(!q.isEmpty()) {
            int front = q.peek();
            q.poll();
            int cnt = 1;
            
            while(!q.isEmpty() && q.peek() <= front) {
                q.poll();
                cnt++;
            }
            
            answer.add(cnt);
        }
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}